test/externalTests/foundry.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2023 solidity contributors.
+# ------------------------------------------------------------------------------
+
+import os
+import re
+from pathlib import Path
+from shutil import which
+from string import Template
+from typing import Tuple
+import requests
+
+from external_test import AVAILABLE_PRESETS
+from external_test import settings_from_preset, run_cmd
+from external_test import TestConfig, TestRunner, ExternalTest
+
+# TODO: Add hardhat runner
+# TODO: Add truffle runner
```

Comment:
Haha, I mean, this should probably go as checklist in the PR description, rather than here, since I doubt you're gonna be adding hardhat and truffle runners inside `foundry.py` :)

Reply:
Indeed. I just added there to not forget the next steps haha
I also need to remove the rust installation todos, since they are being installed by the CI, or we could just use an `cimg-rust` as well.